src/main/java/core/basesyntax/MainApp.java

```diff
@@ -1,5 +1,51 @@
 package core.basesyntax;
+public abstract class Machine {
+  public abstract void doWork();
+  public abstract void stopWork();
+}
 
-public class MainApp {
+public class Truck extends Machine {
+  @Override
+  public void doWork() {
+    System.out.println("Truck started working.");
+}
+  @Override
+  public void stopWork() {
+    System.out.println("Truck stopped working.");
+    } 
+}
 
+public class Bulldozer extends Machine {
+  @Override
+  public void doWork() {
+    System.out.println("Bulldozer started working.");
+  }
+  @Override
+  public void stopWork() {
+    System.out.println("Bulldozer stopped working.");
+    }
 }
+
+public class Excavator extends Machine {
+  @Override
+  public void doWork() {
+    System.out.println("Excavator started working.");
+    }
+  @Override
+  public void stopWork() {
+    System.out.println("Excavator stopped working.");
```

Comment:
Write informative messages in methods. The messages should indicate what type of 'Machine' is working right now 'Truck', 'Bulldozer' or 'Excavator'. The current messages such as 'Truck started working.' are not informative enough. Consider specifying what work the Truck, Bulldozer, or Excavator is doing, for example, 'Truck is transporting cargo.', 'Bulldozer is clearing the construction site.', 'Excavator is digging a trench.'